datamol/convert.py

```diff
@@ -386,30 +387,34 @@ def to_df(
             If called once, it will be applied for the newly created dataframe with
             mol in it.
         render_all_df_mol: Whether to render all pandas dataframe mol column as images.
+        n_jobs: Number of jobs for parallelization. Leave to 1 for no
+            parallelization. Set to -1 to use all available cores.
     """
 
     # Init a dataframe
     df = pd.DataFrame()
 
     # Feed it with smiles
     if smiles_column is not None:
-        smiles = [to_smiles(mol) for mol in mols]
+        smiles = dm.parallelized(to_smiles, mols, n_jobs=n_jobs)
         df[smiles_column] = smiles
 
     # Add a mol column
     if mol_column is not None:
         df[mol_column] = mols
 
     # Add any other properties present in the molecule
-    props = [
-        mol.GetPropsAsDict(
+    def _mol_to_prop_dict(mol):
+        return mol.GetPropsAsDict(
             includePrivate=include_private,
             includeComputed=include_computed,
         )
-        for mol in mols
-    ]
-    props_df = pd.DataFrame(props)
 
+    # EN: You cannot process here because all properties will be lost
+    # An alternative would be https://www.rdkit.org/docs/source/rdkit.Chem.PropertyMol.html
+    # But this has less overhead
+    props = dm.parallelized(_mol_to_prop_dict, mols, n_jobs=n_jobs, scheduler="threads")
```

Comment:
Should we propagate `scheduler`? Have you tested the difference in speed?

Reply:
I guess `progress` should also be propagated no?

Reply:
It can only be `threads` here, as `processes`  wouldn't work. The default serialization done by RDKIT removes all molecular property information. 

Alternative is to convert all molecules to `PropertyMol`, but that conversion can only be done in `threads` too...

Made comment a bit more clear, so we can remember why we are forcing `threads` there.

Reply:
Wondering what would be the difference between working with `PropertyMol` all the time instead of `Chem.rdchem.Mol`

Reply:
> It can only be `threads` here, as `processes` wouldn't work. The default serialization done by RDKIT removes all molecular property information.
> 
> Alternative is to convert all molecules to `PropertyMol`, but that conversion can only be done in `threads` too...
> 
> Made comment a bit more clear, so we can remember why we are forcing `threads` there.

Ok got it now. Thanks for editing your comment in the code.

Reply:
> Wondering what would be the difference between working with `PropertyMol` all the time instead of `Chem.rdchem.Mol`

I didn't know about that. I am open to explore it. What pros do you see?

Reply:
Pros:

- Things like `copy`, pickling, etc will keep all the Props natively
-  mainly will have to change only `to_mol`
- subclass of `dm.Mol`, so refactoring can be easy since `isinstance(PropertyMol(x), dm.Mol)` remains True
- Property mols are more permissive about the types of property values. 

Cons:
- Overheads ? (would need to test this) 
- Would need to check that  `PropertyMol` is a valid substitute to `dm.Mol`  in RDKit's native functions.

Reply:
Thanks. It looks nice indeed.

> Would need to check that PropertyMol is a valid substitute to dm.Mol in RDKit's native functions.

Clearly my biggest concern here.